environment/src/main/java/com/sequenceiq/environment/environment/validation/validators/EncryptionKeyArnValidator.java

```diff
@@ -100,9 +102,12 @@ public ValidationResult validate(EnvironmentValidationDto environmentValidationD
         try {
             CloudEncryptionKeys encryptionKeys =  retryService.testWith2SecDelayMax15Times(() -> cloudPlatformConnectors.get(cloudPlatformVariant).
                     platformResources().encryptionKeys(extendedCloudCredential, region, Collections.emptyMap()));
-            if (encryptionKeys.getCloudEncryptionKeys().stream().map(CloudEncryptionKey::getName)
-                    .noneMatch(s -> s.equals(encryptionKeyArn))) {
-                validationResultBuilder.error("The provided encryption key does not exist in the given region's encryption key list for this credential.");
+            List<String> keyArns = encryptionKeys.getCloudEncryptionKeys().stream().map(CloudEncryptionKey::getName).collect(Collectors.toList());
+            if (keyArns.stream().noneMatch(s -> s.equals(encryptionKeyArn))) {
+                validationResultBuilder.error("Following encryption keys are retrieved from the cloud " + keyArns +
```

Comment:
(nit) ` .The provided encryption` you could change this to `. The provided encryption`
Can you please check `validationResultBuilder` if there are any implicit buffer size limitations for the message size? Thanks.